Sources/DeepLearning/Operators/Basic.swift

```diff
@@ -578,12 +578,12 @@ public extension PartialRangeThrough where Bound == Int {
 
 public extension Tensor {
     @_fixed_layout @usableFromInline
-    internal struct IndexPath {
+    internal struct IndexPath: Differentiable {
```

Comment:
Hmm why is this necessary? It doesn’t make much mathematical sense.

Reply:
It's a hack to make `subscript(_ ranges: TensorRangeExpression...) -> Tensor` differentiable. We could avoid it by allowing something like `.withoutDerivative()` on `IndexPath` maybe.

Reply:
What's the error message if you just mark the original function as `@differentiable(wrt: self)` without making `IndexPath` differentiable? I don't think the workaround is a good direction, and i think there's a way to fix it with smaller workarounds.

Reply:
I managed to work around this by using thunks like this:
```swift
self[{IndexPath({ranges.map { $0.tensorRange }}())}()]
```
However, this is pretty ugly and counterintuitive. We should probably find a better way to deal with non-differentiable parts.